.github/actions/setup_python/action.yml

```diff
@@ -54,13 +53,11 @@ runs:
       with:
         python-version: ${{ inputs.version }}
 
-    - if: ${{ runner.os == 'macOS' || runner.os == 'Windows' || (runner.os == 'Linux' && runner.arch != 'ARM64' && steps.check_python.outputs.installed == 'false' ) }}
+    - if: ${{ runner.os == 'macOS' || runner.os == 'Windows' || (runner.os == 'Linux' && runner.arch != 'ARM64' && steps.check_python.outputs.installed == 'false') }}
       name: Setup Python ${{ inputs.version }}
       uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
       with:
         python-version: ${{ inputs.version }}
-      env:
-        PIP_CACHE_DIR: ${{ inputs.self-hosted-runner == 'true' && inputs.pip-cache-path || '' }}
```

Comment:
as we do not the pip's version the action is going to install, we should not just pass the `inputs.pip-cache-path` because it can corrupt the cache in there.